core/templates/pages/skill-editor-page/skill-editor-page.component.ts

```diff
@@ -25,26 +25,63 @@ require(
 require(
   'pages/skill-editor-page/editor-tab/skill-editor-main-tab.directive.ts');
 require('pages/skill-editor-page/navbar/skill-editor-navbar.directive.ts');
+require(
+  'pages/skill-editor-page/skill-preview-tab/skill-preview-tab.component.ts');
 require(
   'pages/skill-editor-page/navbar/skill-editor-navbar-breadcrumb.directive.ts');
 require(
   'pages/skill-editor-page/questions-tab/skill-questions-tab.directive.ts');
-
+require('domain/editor/undo_redo/undo-redo.service.ts');
+require('domain/utilities/url-interpolation.service.ts');
 require('pages/skill-editor-page/skill-editor-page.constants.ajs.ts');
 require('pages/interaction-specs.constants.ajs.ts');
 
 angular.module('oppia').component('skillEditorPage', {
   template: require('./skill-editor-page.component.html'),
   controller: [
-    'SkillEditorRoutingService', 'SkillEditorStateService', 'UrlService',
+    '$uibModal', 'SkillEditorRoutingService', 'SkillEditorStateService',
+    'UndoRedoService', 'UrlInterpolationService', 'UrlService',
     function(
-        SkillEditorRoutingService, SkillEditorStateService, UrlService) {
+        $uibModal, SkillEditorRoutingService, SkillEditorStateService,
+        UndoRedoService, UrlInterpolationService, UrlService) {
       var ctrl = this;
       ctrl.getActiveTabName = function() {
         return SkillEditorRoutingService.getActiveTabName();
       };
+      ctrl.selectMainTab = function() {
+        SkillEditorRoutingService.navigateToMainTab();
+      };
+      ctrl.selectPreviewTab = function() {
+        SkillEditorRoutingService.navigateToPreviewTab();
+      };
+      ctrl.selectQuestionsTab = function() {
+        // This check is needed because if a skill has unsaved changes to
+        // misconceptions, then these will be reflected in the questions
+        // created at that time, but if page is refreshed/changes are
+        // discarded, the misconceptions won't be saved, but there will be
+        // some questions with these now non-existent misconceptions.
+        if (UndoRedoService.getChangeCount() > 0) {
+          $uibModal.open({
+            templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+              '/pages/skill-editor-page/modal-templates/' +
+              'save-pending-changes-modal.directive.html'),
+            backdrop: true,
+            controller: 'ConfirmOrCancelModalController'
+          }).result.then(null, function() {
```

Comment:
Why pass null for the success callback?

Reply:
It's just an info modal, no action is required.